src/program/latex/mod.rs

```diff
@@ -0,0 +1,1291 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
+use crate::expression::Expression;
+use crate::instruction::{Gate, GateModifier, Instruction, Qubit};
+use crate::Program;
+
+/// Available commands used for building circuits with the same names taken
+/// from the Quantikz documentation for easy reference. LaTeX string denoted
+/// inside `backticks`.
+///
+/// # Available Commands
+///
+///   Single wire commands: lstick, gate, phase, super, qw, nr
+///   Multi-wire commands: ctrl, targ
+#[derive(Clone, Debug)]
+enum Command {
+    /// `\lstick{\ket{q_{u32}}}`: Make a qubit "stick out" from the left.
+    Lstick(String),
+    /// `\gate{name}`: Make a gate on the wire.
+    Gate(String),
+    /// `\phase{symbol}`: Make a phase on the wire with a rotation
+    Phase(String),
+    /// `^{\script}`: Add a superscript to a gate
+    Super(String),
+    /// `\qw`: Connect the current cell to the previous cell i.e. "do nothing".
+    Qw,
+    /// `\\`: Start a new row
+    Nr,
+    /// `\ctrl{wire}`: Make a control qubit--different from Control.
+    Ctrl(String),
+    /// `\targ{}`: Make a controlled-not gate.
+    Targ,
+}
+
+impl Command {
+    fn get_command(command: Self) -> String {
+        match command {
+            Self::Lstick(wire) => format!(r#"\lstick{{\ket{{q_{{{wire}}}}}}}"#),
+            Self::Gate(name) => format!(r#"\gate{{{name}}}"#),
+            Self::Phase(symbol) => format!(r#"\phase{{{symbol}}}"#),
+            Self::Super(script) => format!(r#"^{{\{script}}}"#),
+            Self::Qw => r"\qw".to_string(),
+            Self::Nr => r"\\".to_string(),
+            Self::Ctrl(wire) => format!(r#"\ctrl{{{wire}}}"#),
+            Self::Targ => r"\targ{}".to_string(),
+        }
+    }
+}
+
+/// Types of parameters passed to commands.
+#[derive(Clone, Debug)]
+enum Parameter {
+    /// Symbolic parameters
+    Symbol(Symbol),
+}
+
+impl ToString for Parameter {
+    fn to_string(&self) -> String {
+        match self {
+            Parameter::Symbol(symbol) => Symbol::to_string(symbol),
+        }
+    }
+}
+
+/// Supported Greek and alphanumeric symbols.
+#[derive(Clone, Debug)]
+enum Symbol {
+    Alpha,
+    Beta,
+    Gamma,
+    Phi,
+    Pi,
+    Text(String),
+}
+
+impl ToString for Symbol {
+    fn to_string(&self) -> String {
+        match self {
+            Symbol::Alpha => r"\alpha".to_string(),
+            Symbol::Beta => r"\beta".to_string(),
+            Symbol::Gamma => r"\gamma".to_string(),
+            Symbol::Phi => r"\phi".to_string(),
+            Symbol::Pi => r"\pi".to_string(),
+            Symbol::Text(text) => format!(r#"\text{{{text}}}"#),
+        }
+    }
+}
+
+impl Symbol {
+    /// Returns the supported Symbol variant from text, otherwise, stores the
+    /// unsupported symbol as text in the Text variant.
+    ///
+    /// # Arguments
+    /// `text` - a String representing a Greek or alphanumeric symbol
+    fn match_symbol(text: String) -> Symbol {
```

Comment:
This should be a `impl From<String> for Symbol`, then wherever you use `Symbol::match_symbol(text)` would become `text.into()`.